### PR TITLE
Use iterable for roles instead of array

### PR DESCRIPTION
**Pull request by internalsystemerror**
To allow getUserRoles to return an object that is traversable.

See: https://github.com/zendframework/zend-expressive-authentication/issues/26

### CODE REVIEW

File: src/UserInterface.php [hunk 1]
```diff
@@ -21,5 +21,5 @@ public function getIdentity() : string;
      *
      * @return string[]
      */
-    public function getUserRoles() : array;
+    public function getUserRoles() : iterable;
```
Comment (michalbundyra) [inline src/UserInterface.php:24]: It should be nice to add some test to cover that change.

Comment (internalsystemerror) [reply src/UserInterface.php:24]: What do you propose? No current test for this interface exists that I can tell, but I'm happy to push more commits.

**Comment by ezimuel** [comment]
@internalsystemerror thanks, I resolved some conflicts due to PR outdated. Sorry for the delay.

**Comment by internalsystemerror** [comment]
@ezimuel ty, that will allow the use of doctrine collections now

**Pull request merged**